[TAG] Add localStorage value for enabling TAG & SQL editor

ryanclark opened the pull request:
This adds a new key into the local storage enum for enabling/disabling TAG from showing up in the navigation, as well as enabling/disabling the manual SQL editor.

ryanclark committed:
Add localStorage value for enabling TAG

ryanclark committed:
Add key for enabling the manual SQL editor

jakule commented:
@ryanclark Are we fine merging this PR?

justinas reviewed (changes requested):
See review for usage of the main flag here https://github.com/gravitational/teleport.e/pull/2541#pullrequestreview-1734777827

Access to TAG UI is now controlled by RBAC / Features like all other items, so we need to remove the main flag, and only leave SQL behind the manually-set dev flag.

justinas commented:
> See review for usage of the main flag here [gravitational/teleport.e#2541 (review)](https://github.com/gravitational/teleport.e/pull/2541#pullrequestreview-1734777827)
> 
> Access to TAG UI is now controlled by RBAC / Features like all other items, so we need to remove the main flag, and only leave SQL behind the manually-set dev flag.

Disregard, we intentionally added the flag to **ship without the UI enabled at first** (just the backend pieces).

jakule committed:
Add access graph features retrieval in UserService

The code adds a `fetchAccessGraphFeatures` in UserService that retrieves and stores the access graph features from the provided API URL into the local storage. It also includes error handling to log the error and continue if fetching access graph features fails. The changes also touch the config file to include the API URL for fetching the access graph features.
